mempool/src/shared_mempool.rs

```diff
@@ -479,35 +471,43 @@ async fn process_consensus_request<V>(
 
             let transactions = txns.drain(..).map(SignedTransaction::into).collect();
 
-            MempoolResponse::GetBlockResponse { transactions }
+            (MempoolResponse::GetBlockResponse { transactions }, callback)
         }
-        MempoolRequest::CommitTransactionsRequest {
-            transactions,
-            block_timestamp_usecs,
+        MempoolRequest::CommitNotification {
+            old_version,
+            new_version,
+            callback,
         } => {
-            let mut pool = smp.mempool.lock().unwrap();
-            for transaction in transactions.iter() {
-                pool.remove_transaction(
-                    &transaction.sender,
-                    transaction.sequence_number,
-                    transaction.is_rejected,
-                );
-            }
-
-            if block_timestamp_usecs > 0 {
-                pool.gc_by_expiration_time(Duration::from_micros(block_timestamp_usecs));
+            let committed_txns = smp
+                .storage_read_client
+                .get_transactions(
```

Comment:
can we get rid of storage read here?
StateSync could know and pass that operation by itself